File: 0071-simplify-path/0071-simplify-path.java
class Solution {
    public String simplifyPath(String path) {
        ArrayList<String>stack =  new ArrayList<>();
        String current = "";
        path += "/";
        for(int i=0; i<path.length(); i++) {
            if (path.charAt(i) == '/') {

                if(current.equals("..")) {
                    if(stack.size() > 0) {
                        stack.remove(stack.size() - 1);}
                }else if(!current.equals(".") && current.length() != 0) {
                    stack.add(current);    
                }
                current = "";
            }else {
                current += path.charAt(i);
            }
        }
        if(stack.size() == 0) {return "/";}
        
        String output = "";
        for(String dir: stack) {
            output += ("/" + dir);
        }
        return output;
    }
}

File: 0071-simplify-path/0071-simplify-path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        if path == "" or len(path) == 1:
            return "/"
        i = 0
        stack = []
        if path[0] == "/":
            i += 1
        while i < len(path):
            current = ""
            while i < len(path) and path[i] != "/":
                current += path[i]
                i += 1
            if current  == "..":
                if stack: 
                    stack.pop()
            elif current == ".":
                continue
            elif current != "":
                stack.append(current)
            i += 1
        if not stack:
            return "/"
        res = ""
        for dir in stack:
            res += ("/" + dir)
        return res
            
        
        